torch/export/_trace.py

```diff
@@ -232,19 +232,37 @@ def lookup_fake(x):
     return fake_args, fake_kwargs, fake_mode
 
 
-def _replace_param_buffer_names(param_buffer_table, sig):
+def _replace_param_buffer_names(param_buffer_table, sig, mod, original_state_dict):
+    # Some moules have state_dict hooks, which modifies the state_dict to be different
+    # from the module hierarchy. We find a matching between the state_dict with hook
+    # and the state_dict hook.
+
+    state_dict_keys_map = {}
+
+    if mod._state_dict_hooks or mod._state_dict_pre_hooks:
+        state_dict_without_hook = mod.state_dict(keep_vars=True, use_hooks=False)
```

Comment:
I think we could add a context manager around this call that removes/puts back the state dict hooks on mod upon enter/exit, that way we don't have to make the nn.Module changes